include/ada/url.h

```diff
@@ -165,6 +164,11 @@ namespace ada {
      */
     void set_protocol(const std::string_view input);
 
+    /**
+     * @see https://url.spec.whatwg.org/#dom-url-href
+     */
+    void set_href(const std::string_view input);
```

Comment:
I think that this should return a bool indicating whether it was a success. Certainly, if the provided input is not a valid URL string, then you'd want to return false and leave your url unchanged? At a minimum?

Reply:
I'll definitely add this.

Reply:
Added it.